### PR TITLE
add sngl framework needed for psd rescaling

### PR DESCRIPTION
**Pull request by ahnitz**
This adds the boiler plate functions needed to apply rescaling in larger statistics.

### CODE REVIEW

File: pycbc/events/ranking.py [hunk 1]
```diff
@@ -4,6 +4,49 @@
 import numpy
 
 
+def newsnr_sgveto_psdvar_scaled(snr, bchisq, sgchisq, psd_var_val):
+    """ Combined SNR derived from NewSNR, Sine-Gaussian Chisq and PSD
+    variation statistic """
+    nsnr = numpy.array(newsnr_sgveto(snr, bchisq, sgchisq), ndmin=1)
+    psd_var_val = numpy.array(psd_var_val, ndmin=1)
+    # 1.2 is the expected maximum psd_var_val over gaussian noise.
```
Comment (spxiwh) [inline pycbc/events/ranking.py]: Would be nice to expose the hard-code parameters, even if only as kwargs here.

File: pycbc/events/ranking.py [hunk 2]
```diff
@@ -4,6 +4,49 @@
...
+    nsnr = nsnr / (psd_var_val)**0.33
```
Comment (spxiwh) [inline pycbc/events/ranking.py]: Both the 1.2 and 0.33 numbers

File: pycbc/events/ranking.py [hunk 3]
```diff
@@ -4,6 +4,49 @@
...
+    if hasattr(snr, '__len__'):
+        return nsnr
+    else:
+        return nsnr[0]
+
+def newsnr_sgveto_psdvar_scaled_threshold(snr, bchisq, sgchisq, psd_var_val):
+    """ Combined SNR derived from NewSNR and Sine-Gaussian Chisq"""
+    nsnr = newsnr_sgveto_psdvar_scaled(snr, bchisq, sgchisq, psd_var_val)
+    nsnr = numpy.array(nsnr, ndmin=1)
+    nsnr[bchisq > 2.0] = 1
```
Comment (spxiwh) [inline pycbc/events/ranking.py]: Again, having this hardcoded doesn't seem great.

File: pycbc/events/ranking.py [hunk 4]
```diff
@@ -4,6 +4,49 @@
...
+def newsnr_sgveto_psdvar_com(snr, bchisq, sgchisq, psd_var_val):
+    """ Combined SNR derived from NewSNR, Sine-Gaussian Chisq and PSD
+    variation statistic """
+    nsnr = numpy.array(newsnr_sgveto(snr, bchisq, sgchisq), ndmin=1)
+    psd_var_val = numpy.array(psd_var_val, ndmin=1)
+
+    # 1.2 is the expected maximum psd_var_val over gaussian noise.
+    lgc = psd_var_val >= 1.2
+    nsnr2 = nsnr / (psd_var_val)**0.33
+    nsnr2[lgc] = nsnr[lgc] / psd_var_val[lgc]
```
Comment (spxiwh) [inline pycbc/events/ranking.py]: Hardcoded stuff here too.

**Comment by spxiwh** [comment]
Why can't I see the codeclimate report? Was something significant merged between this branch and master? Maybe rebase!

**Comment by ahnitz** [comment]
@spxiwh Huh?!? Code climate report shows for me.

**Comment by spxiwh** [comment]
Refreshing made codeclimate appear. I think codeclimate complaints are fair: function descriptions should not be identical and then formatting/PEP-8 concerns.

File: pycbc/events/ranking.py (hunk 1 above)
Comment (tdent) [reply pycbc/events/ranking.py]: Seems that the 1.2 number isn't actually used in the stat. So remove the comment

File: pycbc/events/ranking.py (hunk 2 above)
Comment (tdent) [reply pycbc/events/ranking.py]: 0.33 should be set via kwarg I think with a comment that it was empirically tuned on O2 triggers

Comment (tdent) [reply pycbc/events/ranking.py]: Also pep8 please

File: pycbc/events/ranking.py (hunk 4 above)
Comment (tdent) [reply pycbc/events/ranking.py]: Are we going to use this statistic for anything? BTW, I thought the Portsmouth psd stat penalised everything above 1.2 by the square root of psd var.. I'd support having that coded up.

Also what is _com supposed to mean?

Comment (ahnitz) [reply pycbc/events/ranking.py]: I'm removing this. It was a random experiment. 1.2 is already coded and on masater as newsnr_sgveto_psdvar (further down file).

**Comment by ahnitz** [comment]
@spxiwh I've rebased and now travis passes. I don't know how to resolve the remaining CC issues.

**Pull request merged**